src/eyeTracker.m

```diff
@@ -55,161 +30,134 @@
 
-                %% STEP 2
-                % Provide Eyelink with details about the graphics environment
-                % and perform some initializations. The information is returned
-                % in a structure that also contains useful defaults
-                % and control codes (e.g. tracker state bit and Eyelink key values).
+                %% Initialization
+                % Provide Eyelink with details about the graphics environment and perform some
+                %  initializations. The information is returned in a structure that also contains
+                %  useful defaults and control codes (e.g. tracker state bit and Eyelink key
+                %  values).
+
+                % Provide Screen id where present the calibration
                 el = EyelinkInitDefaults(cfg.screen.win);
 
-                % calibration has silver background with black targets, sound and smaller
-                % targets
+                % Calibration has silver background with black targets, sound and smaller
+                %  targets
                 el.backgroundcolour = [192 192 192, (cfg.screen.win)];
                 el.msgfontcolour = BlackIndex(cfg.screen.win);
                 el.calibrationtargetcolour = BlackIndex(cfg.screen.win);
                 el.calibrationtargetsize = 1;
                 el.calibrationtargetwidth = 0.5;
                 el.displayCalResults = 1;
 
-                % call this function for changes to the calibration structure to take
-                % affect
+                % Call this function for changes to the calibration structure to take
+                %  affect.
                 EyelinkUpdateDefaults(el);
 
-                %% STEP 3
-                % Initialization of the connection with the Eyelink Gazetracker.
-                % exit program if this fails.
-
-                % Initialize EL and make sure it worked: returns: 0 if OK, -1 if error
-                ELinit = Eyelink('Initialize');
+                % Initialize EL and make sure it worked: returns 0 if OK, -1 if error.
+                %  Exit program if this fails.
+                elInit  = Eyelink('Initialize');
                 if ELinit ~= 0
                     fprintf('Eyelink is not initialized, aborted.\n');
                     Eyelink('Shutdown');
-                    Screen('CloseAll');
+                    CleanUp();
                     return
                 end
 
-                % Make sure EL is still connected: returns 1 if connected,
-                % -1 if dummy-connected,
-                % 2 if broadcast-connected and 0 if not connected
-                ELconnection = Eyelink('IsConnected');
+                % Make sure EL is still connected: returns 1 if connected, -1 if dummy-connected,
+                %  2 if broadcast-connected and 0 if not connected. Exit program if this fails.
+                elConnection = Eyelink('IsConnected');
                 if ELconnection ~= 1
```

Comment:
I would refactor those 2 into on single function `eyetrackerCheckConnection` and make those messages either warning or errors.

The advantage is that if there is an error we will land in the `catch` section of the main script and you can remove the calls to `cleanUp` that we have here and also get rid of the `return`.

Reply:
awesome

thanks